### PR TITLE
[MRG] Clean up "introduction" text in docs

### PR DESCRIPTION
**Pull request by standage**
Doing my best to bring this up-to-date. See #853.

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

File: doc/introduction.rst [hunk 1]
```diff
@@ -43,96 +43,59 @@ Introduction to khmer
...
-
  - eliminating reads that will not be used by a de Bruijn graph assembler;
-
  - removing reads with low- or high-abundance k-mers;
-
  - trimming reads of certain kinds of sequencing errors;
-
  - counting k-mers and estimating data set coverage based on k-mer counts;
-
  - running Velvet and calculating assembly statistics;
```
Comment (standage) [inline doc/introduction.rst]: Really? Is this still true?

File: doc/introduction.rst [hunk 2]
```diff
@@ -43,96 +43,59 @@ Introduction to khmer
...
-The most important thing to think about when using khmer is whether or not the
-transformation or filter you're applying is appropriate for the data you're
-trying to assemble.  Two of the most powerful operations available in khmer,
-graph-size filtering and graph partitioning, only make sense for assembly
-datasets with many theoretically unconnected components.  This is typical of
-metagenomic data sets.
+The most important thing to think about when using khmer is whether or not the transformation or filter you're applying is appropriate for the data you're trying to assemble.
+For example, two of the most powerful operations available in khmer, graph-size filtering and graph partitioning, only make sense for assembly datasets with many theoretically unconnected components.
+This is typical of metagenomic data sets.
+Also, while digital normalization can be helpful for transcriptome *assembly*, it is inappropriate for other RNA-seq applications, such as differential expression analysis, that rely on signal from variable coverage.
```
Comment (standage) [inline doc/introduction.rst]: Too soon?

File: doc/introduction.rst [hunk 3]
```diff
@@ -43,96 +43,59 @@ Introduction to khmer
...
-
  - trimming reads of certain kinds of sequencing errors;
-
  - counting k-mers and estimating data set coverage based on k-mer counts;
-
  - running Velvet and calculating assembly statistics;
-
  - optimizing assemblies on various parameters;
-
  - converting FASTQ to FASTA;
```
Comment (standage) [inline doc/introduction.rst]: Could also mention interleaving/decoupling fastq.

**Comment by standage** [comment]
This is ready for review. I'm guessing @ctb will want a crack at this.

**Comment by standage** [comment]
\*bump\*

**Comment by standage** [comment]
@betatim and @luizirber and I are going to hack a bit on khmer at PyCon (and @camillescott, if she's  coming and interested). If @ctb has not reviewed since then I'll recruit someone else to review this then. Hopefully we can check all of the remaining 2.1 release boxes before PyCon is over.

**Comment by ctb** [comment]
What's my deadline on this? I can get to it Sat morning for sure, Friday if it's urgent.

Then at PyCon y'all could release khmer 2.1, and maybe get the new Cython stuff (#1595) merged as well as #1590.

**Comment by standage** [comment]
No specific sense of urgency, just didn't want the PR (and thus the release) to languish for an indeterminate amount of time. It can definitely wait until Saturday. :-)

File: doc/introduction.rst (hunk 1 above)
Comment (ctb) [inline doc/introduction.rst]: indeed it is not true :)

**Pull request merged**